pkg/cmd/util/clientcmd/factory.go

```diff
@@ -452,6 +462,16 @@ func getPorts(spec api.PodSpec) []string {
 	return result
 }
 
+func getProtocols(spec api.PodSpec) map[string]string {
+	result := make(map[string]string)
+	for _, container := range spec.Containers {
+		for _, port := range container.Ports {
+			result[strconv.Itoa(int(port.ContainerPort))] = string(port.Protocol)
```

Comment:
Same code as upstream expose.

On Wed, Jul 6, 2016 at 10:06 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/util/clientcmd/factory.go
> <https://github.com/openshift/origin/pull/9578#discussion_r69736152>:
>
> > @@ -452,6 +462,16 @@ func getPorts(spec api.PodSpec) []string {
> >  	return result
> >  }
> >
> > +func getProtocols(spec api.PodSpec) map[string]string {
> > +	result := make(map[string]string)
> > +	for _, container := range spec.Containers {
> > +		for _, port := range container.Ports {
> > +			result[strconv.Itoa(int(port.ContainerPort))] = string(port.Protocol)
>
> Aren't I allowed to duplicate a port.ContainerPort value with different
> protocols?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9578/files/ca26a48ee495d1c8125d40d8e8a84a5484b6b23c..50b1443fa9c7856958db9948a9be4052949eedfe#r69736152>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p-nrFazlySfPedntCb30ulTzLgolks5qS7ZWgaJpZM4I-xTE>
> .
>